handout/http/catching-rejections/cancel_request.md

```diff
@@ -0,0 +1,25 @@
+## Cancel a Request
+
+Cancelling an HTTP request is a common requirement. For example, you could have a queue of requests where a new request supersedes a pending request and that pending request needs to be cancelled.
+
+To cancel a request we call the `unsubscribe` function of its subscription.
+
+```ts
+@Component({ ... })
+export class MyApp {
+	...
+
+	search() {
+		const request = this.searchService.search(this.searchField.value)
+		  .subscribe(
+		    (result) => { this.result = result.artists.items; },
+		    (err) => { this.errorMessage = err.message; },
+        () => { console.log('Completed'); }
+		  );
+
+		request.unsubscribe();
```

Comment:
I've notice that in the plunker you have this instead:

```ts
setTimeout(() => {
  request.unsubscribe();
  this.errorMessage = 'Request cancelled';
}, 0)
```

Why the `setTimeout` set to 0?